### ISSUE TITLE
Epic: other codes

### ISSUE THREAD

**Issue by aimalz** [LSSTDESC/clmassmod#4]
Let's use this as a space to collect issues about other codes:
What other codes do we want to interface with?
What purposes will they serve in this project and what modules do they belong in?
How do we work them into this code?

**Comment by cavestruz** [comment]
Other possible codes:
cluster_toolkit
CCL
colossus

**Comment by pennalima** [comment]
Hi all,

I'm implementing and extending Douglas's code in NumCosmo. 
https://numcosmo.github.io/
https://github.com/NumCosmo/NumCosmo

I discussed this project on 2017-09-05: https://docs.google.com/document/d/1ev6g4IwzRBpMNCMmYE2_lb-9Wo8o_mVhDLOuLq3_I8E/edit#

Best,
Mariana

**Comment by aimalz** [comment]
If the goals of clmassmod and NumCosmo are the same, then there's not much reason to have separate projects.  Based on looking at the NumCosmo examples, I think there could still be value in #3, but it's unclear what should be underneath the hood.  @pennalima What is the state of NumCosmo's implementation of @deapplegate's code's functionality?  @cavestruz Were there goals beyond wrapping a flexible, modular package that does what @deapplegate's code does?

**Comment by cavestruz** [comment]
Hi mariana @pennalima  - I'm taking a look at the notes from the telecon, and just pulled up your [slides](https://drive.google.com/file/d/0B7iOQiNE_OjvSTBURVFkSUxnT2s/view).  I think I was absent that call, so I actually wasn't aware that you were undertaking a refactoring in NumCosmo.  Correct me if I'm missing something, but it sounds like the biggest extension is incorporating mass modeling profiles beyond NFW?   Are you implementing all of Doug's code functionality?  

@aimalz - Beyond a flexible modular package for @deapplegate's code, further goals (not near term) would be to have more flexibility in how the mass and concentration (or other mass model parameters) are fit.  But, the flexible modular package was a primary initial goal.

**Comment by pennalima** [comment]
Yes @cavestruz, I am implementing all of Doug's code functionality. Actually, I've already implemented some of them previously. 

@aimalz For now I implemented almost all the "basic" theoretical functions to compute the lensing quantities. The work is progressing slowly now because I'm in maternity leave.  

@aimalz @cavestruz @tmcclintock NumCosmo presents a modular structure that it can be very useful to deal with some issues discussed in #3 . NumCosmo can be linked to clmassmod likewise GSL, for example.
